nfts/src/lib.rs

```diff
@@ -516,10 +534,15 @@ impl<T: Config> traits::NFTTrait for Pallet<T> {
 	fn set_viewer(id: NFTId, value: Option<Self::AccountId>) -> DispatchResult {
 		Data::<T>::try_mutate(id, |maybe_data| -> DispatchResult {
 			let data = maybe_data.as_mut().ok_or(Error::<T>::NFTNotFound)?;
-			data.viewer = value;
-			Ok(())
+			data.is_delegated = value.is_some();
+			Ok(().into())
 		})?;
 
+		match value.as_ref() {
```

Comment:
I think that you don't need as_ref here.

Reply:
You are right ! Done